README.md

```diff
@@ -1,48 +1,207 @@
+      -p 8080:8080 \
+      absaoss/spline-rest-server
+```
+
+Or you can deploy it as a WAR-file into any Java compatible Web-Container, e.g. Tomcat.
+You can find a WAR-file in the Maven repo here:
+[```za.co.absa.spline:rest-gateway:0.4.0```](https://repo1.maven.org/maven2/za/co/absa/spline/rest-gateway/0.4.0/)
+
+Add the argument for the arango connection string `-Dspline.database.connectionUrl=arangodb://localhost/spline`
+
+The server exposes the following REST API:
+-   Producer API (`/producer/*`) 
+-   Consumer API (`/consumer/*`)
+
+... and other useful URLs:
+-   Running server version information: [/about/version](http://localhost:8080/about/version)
+-   Producer API Swagger documentation: [/docs/producer.html](http://localhost:8080/docs/producer.html) 
+-   Consumer API Swagger documentation: [/docs/consumer.html](http://localhost:8080/docs/consumer.html) 
+
+## Start Spline UI
+
+Spline web client can be started using 3 diffrent ways:
+
+-   Docker: 
+
+```shell script
+docker container run \
+      -e spline.consumer.url=http://172.17.0.1:8080/consumer \
+      -p 9090:8080 \
+      absaoss/spline-web-client
+```
+
+-   Java compatible Web-Container:
+
+You can find the WAR-file of the Web Client in the repo here:
+[```za.co.absa.spline:client-web:0.4.0```](https://repo1.maven.org/maven2/za/co/absa/spline/client-web/0.4.0/)
+
+Add the argument for the consumer url `-Dspline.consumer.url=http://localhost:8080/consumer`
+
+-   Node JS application: 
+
+Download [```node.js```](https://nodejs.org/en/) then install [```@angular/cli```](https://www.npmjs.com/package/@angular/cli) to run `ng serve` or `ng-build` command.
+
+To specify the consumer url please edit the [config.json](https://github.com/AbsaOSS/spline/blob/develop/client-ui/src/assets/config.json) file
+
+You can find the documentation of this module in [ClientUI](client-ui/README.md).
+
+## Check the result in the browser
+<http://localhost:9090>
+
+## Use spline in your application
+Add a dependency on Spark Agent.
+```xml
+<dependency>
+    <groupId>za.co.absa.spline</groupId>
+    <artifactId>spark-agent</artifactId>
+    <version>0.4.0</version>
+</dependency>
+```
+In your spark job you have to enable spline.
+```scala
+// given a Spark session ...
+val sparkSession: SparkSession = ???
+
+// ... enable data lineage tracking with Spline
+import za.co.absa.spline.harvester.SparkLineageInitializer._
+sparkSession.enableLineageTracking()
+
+// ... then run some Dataset computations as usual.
+// Data lineage of the job will be captured and stored in the
+// configured database for further visualization by Spline Web UI
+```
+### Properties
+
+You also need to set some configuration properties. Spline combine these properties from several sources:
+1.  Hadoop config (`core-site.xml`)
+
+2.  JVM system properties
+
+3.  `spline.properties` file in the classpath
+
+#### `spline.mode`
+-   **DISABLED** Lineage tracking is completely disabled and Spline is unhooked from Spark.
```

Comment:
Tiny: Typography of this seems off. Using *`DISABLED`* feels more pleasing to eyes.